pilot/cmd/pilot-agent/app/wait.go

```diff
@@ -41,17 +42,23 @@ var (
 			log.Infof("Waiting for Envoy proxy to be ready (timeout: %d seconds)...", timeoutSeconds)
 
 			var err error
-			timeoutAt := time.Now().Add(time.Duration(timeoutSeconds) * time.Second)
-			for time.Now().Before(timeoutAt) {
-				err = checkIfReady(client, url)
-				if err == nil {
-					log.Infof("Envoy is ready!")
-					return nil
+			go func() {
+				time.Sleep(time.Duration(timeoutSeconds) * time.Second)
+				timeoutCheck <- struct{}{}
+			}()
+			for {
+				select {
+				case <-timeoutCheck:
+					return fmt.Errorf("timeout waiting for Envoy proxy to become ready. Last error: %v", err)
+				default:
+					err = checkIfReady(client, url)
```

Comment:
Then it is a dead loop calling `checkIfReady`?

Reply:
Yeah, but both timeout and non error check path can stop this loop.

Reply:
I am concerned with the high cpu usage maybe

Reply:
yeah, however, as you know, `pilot-agent wait` is running via k8s `postStart` mechanism, so application container is suspended during this wait command is not over. So the CPU usage may not be the key point.

Reply:
This still wastes node cpu

Reply:
so @hzxuzhonghu & @howardjohn,  as mentioned above, using `time.After` channel in select would be a better choice. I will submit again!